flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroSchemaConvertor.java

```diff
@@ -200,4 +246,220 @@ private static DataType handleLogicalTypeSchema(Schema fieldSchema) {
         }
         return null;
     }
+
+    /**
+     * Converts Flink SQL {@link LogicalType} (can be nested) into an Avro schema.
+     *
+     * <p>Use "org.apache.flink.avro.generated.record" as the type name.
+     *
+     * @param schema the schema type, usually it should be the top level record type, e.g. not a
+     *     nested type
+     * @return Avro's {@link Schema} matching this logical type.
+     */
+    public static Schema convertToSchema(LogicalType schema) {
+        return convertToSchema(schema, "org.apache.flink.avro.generated.record");
+    }
+
+    /**
+     * Converts Flink SQL {@link LogicalType} (can be nested) into an Avro schema.
+     *
+     * <p>The "{rowName}_" is used as the nested row type name prefix in order to generate the right
+     * schema. Nested record type that only differs with type name is still compatible.
+     *
+     * @param logicalType logical type
+     * @param rowName the record name
+     * @return Avro's {@link Schema} matching this logical type.
+     */
+    public static Schema convertToSchema(LogicalType logicalType, String rowName) {
```

Comment:
This is a very long method with a long switch statement. Please breakup this method into smaller methods. You can keep the switch statement in this method but the block of code for the case statements can be moved into separate methods. Especially the time handling, row handling and decimal handling. I see duplication of code between the case statements.

Reply:
Fixed.